src/evaluate/inspect.py

```diff
@@ -31,41 +32,67 @@ class SplitsNotFoundError(ValueError):
     pass
 
 
-def list_metrics(with_community_metrics=True, with_details=False):
+def list_evaluation_modules(type=None, include_community=True, with_details=False):
     """List all the metrics script available on the Hugging Face Hub.
 
     Args:
-        with_community_metrics (:obj:`bool`, optional, default ``True``): Include the community provided metrics.
-        with_details (:obj:`bool`, optional, default ``False``): Return the full details on the metrics instead of only the short name.
+        type (:obj:`string`, optional, default ``None``): Type of evaluation modules to list. If ``None`` all types are listed.
+        include_community (:obj:`bool`, optional, default ``True``): Include community modules in the list.
+        with_details (:obj:`bool`, optional, default ``False``): Return the full details on the metrics instead of only the ID.
+
+    Returns:
+        `list`
     """
-    metrics = huggingface_hub.list_metrics()
-    if not with_community_metrics:
-        metrics = [metric for metric in metrics if "/" not in metric.id]
-    if not with_details:
-        metrics = [metric.id for metric in metrics]
-    return metrics
+    if type is None:
+        evaluations_list = []
+        for type in ["metric", "comparison", "measurement"]:
+            evaluations_list.extend(
+                _list_evaluation_modules_type(type, include_community=include_community, with_details=with_details)
+            )
+    else:
+        evaluations_list = _list_evaluation_modules_type(
+            type, include_community=include_community, with_details=with_details
+        )
+    return evaluations_list
+
+
+def _list_evaluation_modules_type(type, include_community=True, with_details=False):
+
+    r = requests.get(HF_LIST_ENDPOINT.format(type=type))
+    r.raise_for_status()
+    d = r.json()
```

Comment:
At one point it would be cool to have `list_spaces` in `huggingface_hub`. I opened https://github.com/huggingface/huggingface_hub/issues/881